src/envoy/utils/http_header_utils.cc

```diff
@@ -38,19 +38,22 @@ absl::string_view readHeaderEntry(const Envoy::Http::HeaderEntry* entry) {
 
 absl::string_view extractHeader(const Envoy::Http::HeaderMap& headers,
                                 const Envoy::Http::LowerCaseString& header) {
-  const auto* entry = headers.get(header);
-  return readHeaderEntry(entry);
+  const auto result = headers.get(header);
+  if (result.empty()) {
+    return Envoy::EMPTY_STRING;
+  }
+  return readHeaderEntry(result[0]);
```

Comment:
Prefer to wrap with `getAllOfHeaderAsString` to get a string list of all header values instead of only the first one.

Reply:
+1, it would match our previous behavior correct? Previously I believe all duplicate headers were concat into a single string.